plot/steer_aggregate_and_plot_observables.py

```diff
@@ -216,26 +448,121 @@ def main():
+                            continue
+                # check if file final_state_hadron_dir/DONE exist and the downloader finished buffering this run, if it does not exist, then wait until it exists
+                while 1:
+                    if os.path.exists(os.path.join(final_state_hadron_dir, 'DONE')):
+                        break
+                    print(f'Waiting for download of run to finish, re-checking in 60s...')
+                    time.sleep(60)
+
+                from js_stat_xsede_steer import submit
+                
+                # temporary (?) fix as subprocess 
+                arg_list = [f'python3',f'{stat_xsede_2021_dir}/scripts/js_stat_xsede_steer/submit.py',
+                            f'analysis',
+                            f'--run_number={run_number}', 
+                            f'--run_info_dir={run_info_download_location}', 
+                            f'--final_state_hadrons_dir={final_state_hadrons_download_location}', 
+                            f'--output_dir={reananalysis_output_location}', 
+                            f'--analysis_facility_name={local_analysis_facility}', 
+                            f'--download_facility_name={facility}',
+                            f'--container_path={analyis_container_path}', 
+                            f'--n_tasks={ntasks}', 
+                            f'--max_n_slurm_jobs={max_n_slurm_jobs}', 
+                            f'--jetscape_analysis_dir={jetscape_analysis_dir}']
+                subprocess.run(arg_list, check=True, shell=False)
+                if delete_final_state_hadrons_after_analysis:
+                    # safely delete final_state_hadron_dir, make sure the path contains final_state_hadrons_per_run so we don't by accident delete "/" or "/home" .... 
+
+                    if os.path.exists(final_state_hadron_dir) and "final_state_hadrons_per_run" in final_state_hadron_dir:
+                        # remove all files in the directory that contain final_state_hadrons and parquet in name
+                        # note that we don't ask for endging parquet since sometimes incomplete files have weird ending
+                        print(f'Deleting all parquet files in {final_state_hadron_dir} as requested...')
+                        for f in os.listdir(final_state_hadron_dir):
+                            if "final_state_hadrons" in f and "parquet" in f:
+                                os.remove(os.path.join(final_state_hadron_dir, f))
+                            # remove DONE file
+                            if "DONE" in f:
+                                os.remove(os.path.join(final_state_hadron_dir, f))
+                        # now we can delete the directory
+                        os.rmdir(final_state_hadron_dir)
+
+                # TODO upload analysis results / histos to OSN in the future
```

Comment:
I think this is still pending, right? If so, I think we should still commit this PR as is and then add that next. But if it's actually already done, please cleanup this TODO

Reply:
it is still todo and will leave this for future commit